controllers/reviews.js

```diff
@@ -0,0 +1,108 @@
+var Review = require('../models/review');
+var accepts = require('accepts');
+
+/*
+* GET ALL reviews
+* @method GET
+*/
+exports.index = function (req, res, next) {
+    var accept = accepts(req);
+    Review.find(function(err, reviews){
+        if(accept.type(['json', 'html']) == 'json') {
+            res.setHeader('Content-Type', 'application/json')
+            res.status(200).json(reviews);
+        }else {
+            res.setHeader('Content-Type', 'text/html')
+            res.render('reviews/index', {reviews: reviews})
+        }
+    });
+}
+
+/*
+* CREATE a new reviews
+* @method POST
+* @params id
+*/
+exports.create = function (req, res, next) {
+    var review = new Review(req.body);
+    review.save(function(err){
+        if (err) {
+            res.status(400).json('bad review')
+        }
+        res.status(201).json('created');
+    });
+}
+
+exports.edit = function (req, res, next) {
+    var accept = accepts(req);
+    Review.findById(req.params.id, function(err, review){
+        if(err) {
+            res.status(404).send('review not found');
```

Comment:
Pareil, ici err vient de la base, donc c'est plus un souci technique. Par exemple, si la base est hors service, malheursement on enverra un 404 comme status code, et c'est le bon status code.

Cependant, si jamais tu voudrais envoyer un 404 ici (en plus du 500), il faut inspecter err. Un 404 on peut l'avoir quand on fourni un ObjectId que n'a pas la bon syntaxe.  Un ObjectId de Mongo est une chaine avec 15 characters. Si on fait une requete 

http://localhost:3000/api/reviews/edit/54f2eeb9b91fc8df18262217

on passera jamais par ici car l'id est valide, mais n'existe pas, donc err est undefined et review aussi.

si http://localhost:3000/api/reviews/edit/54f4221, là en effet, err aura une valeur comme ceci

{
   "error":{
   "message": "Cast to ObjectId failed for value \"54f4221\" at path \"_id\"",
   "name": "CastError",
  "type": "ObjectId",
  "value": "54f4221",
  "path": "_id"
}
}

et "eventuellement" un 404 pourrait être envoyé.